src/components/fields/category-select-field.vue

```diff
@@ -155,6 +166,26 @@ export default defineComponent({
       isDropdownOpened.value = state ?? !isDropdownOpened.value;
     };
 
+    const filterCategories = (categories, query) => {
+      let result = [];
+
+      for (const category of categories) {
+        if (category.name.includes(query)) {
+          result.push(category);
+        }
+
+        // eslint-disable-next-line vue/max-len
+        if (category.subCategories && category.subCategories.length > 0 && searchValue.value.length) {
+          const filteredSubCategories = filterCategories(category.subCategories, query);
```

Comment:
Instead of recursion, you can take a look into the `categories` store and find there `rawCategories` array. It might be better to use that array like in the example instead of all this recursion.
```js
const filteredItems = computed(() => rawCategories.filter(item => item.name.includes(searchQuery))
```

So that you will also be able to remove that new check from `<template v-if="item.subCategories.length && !searchValue.length">`.

So basically until you have any `searchQuery`, display `levelValues` in the template. Otherwise display `filteredItems`. This condition should be inside the script, not in the template